### PR TITLE
export promQL syntax as a map

### PR DESCRIPTION
**Pull request by Nexucis**
/assigned to @juliusv

### CODE REVIEW

File: src/promql.term.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+export const promqlSyntax = {
+    "MatchOp": [
+        "=",
+        "!=",
+        "=~",
+        "!~"
+    ],
+    "BinaryExpr": [
```
Comment (juliusv) [inline src/promql.term.js]: These are technically not binary expressions, but binary operators, so `BinaryOp` would be a better name.

**Comment by Nexucis** [comment]
should improve the hybrid mode and maybe can be integrated to this PR : https://github.com/prometheus-community/codemirror-promql/pull/11

**Comment by juliusv** [comment]
Hmm, I'm still a bit undecided whether the lezer code should be mixed with this repo, since none of the other Lezer language repos (like https://github.com/lezer-parser/javascript) export extra non-lezer stuff. I guess your primary motivation is to avoid including two different dependencies?

File: src/promql.term.js (hunk 1 above)
Comment (Nexucis) [reply src/promql.term.js]: actually the name here is following the NodeName you put in the grammar:

```
BinaryExpr {
  Expr !pow BinOp<"^">    BinModifier Expr |
  Expr !mul BinOp<"*">    BinModifier Expr |
  Expr !mul BinOp<"/">    BinModifier Expr |
  Expr !mul BinOp<"%">    BinModifier Expr |
  Expr !add BinOp<"+">    BinModifier Expr |
  Expr !add BinOp<"-">    BinModifier Expr |
  Expr !eql BinOp<"==">   BinModifier Expr |
  Expr !eql BinOp<">=">   BinModifier Expr |
  Expr !eql BinOp<">">    BinModifier Expr |
  Expr !eql BinOp<"<">    BinModifier Expr |
  Expr !eql BinOp<"<=">   BinModifier Expr |
  Expr !eql BinOp<"!=">   BinModifier Expr |
  Expr !and ckw<"and">    BinModifier Expr |
  Expr !and ckw<"unless"> BinModifier Expr |
  Expr !or  ckw<"or">     BinModifier Expr
}
```

From my point of view it makes more sense to put `BinaryExpr` as it is the top node that handles every kind of BinOp or "and" / "unless" / "or".
But I can change it if you really think it makes more sense to put `BinaryOp`

**Comment by Nexucis** [comment]
yes, and from my point of view, lezer should allow the user to choose to export some string list for the autocompletion.
I guess everybody would like to autocomplete the keyword of their language.

But maybe that's not the good way to fix this issue and maybe the good way to do it is to use the token.js things. I didn't really look at it actually.

File: src/promql.term.js (hunk 1 above)
Comment (Nexucis) [reply src/promql.term.js]: I changed it.

**Comment by juliusv** [comment]
Yeah, the best would be if Lezer itself exposed this information in a structured way. In the absence of that, I'll merge this for now, although I'm not sure that it should live in this repo long-term.

**Pull request merged**